GameServerCore/Maps/IMapScriptHandler.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Numerics;
+using GameServerCore.Content;
+using GameServerCore.Domain;
+using GameServerCore.Domain.GameObjects;
+using GameServerCore.Enums;
+
+namespace GameServerCore.Maps
+{
+    /// <summary>
+    /// Contains all map related game settings such as collision handler, navigation grid, announcer events, and map properties.
+    /// </summary>
+    public interface IMapScriptHandler : IUpdate
+    {
+        /// <summary>
+        /// Unique identifier for the Map (ex: 1 = Old SR, 11 = New SR)
+        /// </summary>
+        int Id { get; }
+        /// <summary>
+        /// Collision Handler to be instanced by the map. Used for collisions between GameObjects or GameObjects and terrain.
+        /// </summary>
+        ICollisionHandler CollisionHandler { get; }
+        /// <summary>
+        /// Navigation Grid to be instanced by the map. Used for terrain data.
+        /// </summary>
+        INavigationGrid NavigationGrid { get; }
+        IMapData MapData { get; }
+        /// <summary>
+        /// MapProperties specific to a Map Id. Contains information about passive gold gen, lane minion spawns, experience to level, etc.
```

Comment:
Update summary `MapScript which determines functionality for a specific map. Contains information about passive gold gen, lane minion spawns, experience to level, etc.`